components/renku_data_services/users/db.py

```diff
@@ -101,6 +106,21 @@ async def _get_users(self, email: str | None = None) -> List[UserInfo]:
             return [orm.dump() for orm in orms]
 
 
+def create_user_added_message(result: UserInfo, **_) -> UserAdded:
+    """Transform user to message queue message."""
+    return UserAdded(id=result.id, firstName=result.first_name, lastName=result.last_name, email=result.email)
+
+
+def create_user_updated_message(result: UserInfo, **_) -> UserUpdated:
+    """Transform user to message queue message."""
+    return UserUpdated(id=result.id, firstName=result.first_name, lastName=result.last_name, email=result.email)
+
+
+def create_user_removed_message(result, user_id: str) -> UserRemoved:
+    """Transform user removal to message queue message."""
+    return UserRemoved(id=user_id)
```

Comment:
Does it cause problems (with circular imports) if you move these to the `message_queue` module? I.e. at `components/renku_data_services/message_queue`. I think that would be even better.

If we can move these, then it applies to all the other functions being passed to the decorators in the different modules.

Also if you want you can go a step further and make the specific decorators for each case in the `message_queue` module so then the specific decorator is imported and used. Rather than importing the general decorator and the specific event factory and then combining them.

It seems these were already part of the message_queue module and this PR moved them out. I think it is better if they stay in the message_queue module.